### ISSUE TITLE
Underflow while syncing

### ISSUE THREAD

**Issue by rytone** [paritytech/parity#6062]
I cannot find a reliable way to reproduce this.

```
attempt to subtract with overflow', ethcore\src\client\client.rs:115
stack backtrace:
   0: std::sys_common::backtrace::_print
             at C:\projects\rust\src\libstd\sys_common\backtrace.rs:94
   1: std::panicking::default_hook::{{closure}}
             at C:\projects\rust\src\libstd\panicking.rs:354
   2: std::panicking::default_hook
             at C:\projects\rust\src\libstd\panicking.rs:371
   3: std::panicking::rust_panic_with_hook
             at C:\projects\rust\src\libstd\panicking.rs:549
   4: std::panicking::begin_panic<collections::string::String>
             at C:\projects\rust\src\libstd\panicking.rs:511
   5: std::panicking::begin_panic_fmt
             at C:\projects\rust\src\libstd\panicking.rs:495
   6: std::panicking::rust_begin_panic
             at C:\projects\rust\src\libstd\panicking.rs:471
   7: core::panicking::panic_fmt
             at C:\projects\rust\src\libcore\panicking.rs:69
   8: core::panicking::panic
             at C:\projects\rust\src\libcore\panicking.rs:49
   9: ethcore::client::client::{{impl}}::sub
             at E:\more\ethereum\parity_src\ethcore\src\client\client.rs:115
  10: parity::informant::Informant<parity::informant::FullNodeInformantData>::tick<parity::informant::FullNodeInformantData>
             at E:\more\ethereum\parity_src\parity\informant.rs:268
  11: parity::informant::{{impl}}::timeout<parity::informant::FullNodeInformantData>
             at E:\more\ethereum\parity_src\parity\informant.rs:400
  12: ethcore_io::worker::Worker::do_work<ethcore::service::ClientIoMessage>
             at E:\more\ethereum\parity_src\util\io\src\worker.rs:120
  13: ethcore_io::worker::Worker::work_loop<ethcore::service::ClientIoMessage>
             at E:\more\ethereum\parity_src\util\io\src\worker.rs:101
  14: ethcore_io::worker::{{impl}}::new::{{closure}}<ethcore::service::ClientIoMessage>
             at E:\more\ethereum\parity_src\util\io\src\worker.rs:79
  15: std::panic::{{impl}}::call_once<(),closure>
             at C:\projects\rust\src\libstd\panic.rs:296
  16: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
             at C:\projects\rust\src\libstd\panicking.rs:454
  17: panic_abort::__rust_maybe_catch_panic
             at C:\projects\rust\src\libpanic_abort\lib.rs:40
  18: std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
             at C:\projects\rust\src\libstd\panicking.rs:433
  19: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
             at C:\projects\rust\src\libstd\panic.rs:361
  20: std::thread::{{impl}}::spawn::{{closure}}<closure,()>
             at C:\projects\rust\src\libstd\thread\mod.rs:360
  21: alloc::boxed::{{impl}}::call_box<(),closure>
             at C:\projects\rust\src\liballoc\boxed.rs:640
  22: std::sys::imp::thread::{{impl}}::new::thread_start
             at C:\projects\rust\src\libstd\sys\windows\thread.rs:50
  23: BaseThreadInitThunk
```

Parity v1.8.0 (commit 3fb8a85) / Rust 1.18.0 / Windows 10

**Comment by 5chdn** [comment]
Does this also happen on 1.7.0?

**Comment by arkpar** [comment]
This has been fixed in 1.7.0 and latest master
